analysis/tests/src/expected/BrokenParserCases.res.txt

```diff
@@ -0,0 +1,62 @@
+Dump AST src/BrokenParserCases.res 1:22
+
+Source:
+// let _ = ({someProp: , otherProp: 123}, 123)
+//                    ^ast
+
+<*>Pstr_value(
```

Comment:
What does `<*>` indicate? Seems redundant information-wise, but I guess it follows some existing convention of something.

Reply:
No existing convention, just something I made up. Here's a legend:
- `<*>` means it has the cursor
- `<x>` means it has an empty loc

It's redundant at the top level `Pstr_value` etc because the value itself would never get printed unless the cursor was there. But, I was confused several times myself _not_ seeing `<*>`, so I added it there too. Imo having it there too makes it easier to scan for "anchor points" of the cursor, like "there's no cursor (but a broken loc) inside of the structure item, but the structure item itself has the cursor, so we can use that to figure out the context".

Reply:
Got it. How about the ^ symbol instead of * so it's consistent with other tests.

Also question on terminology: is this a hit test? So everything containing the cursor has it, not a single node?

Reply:
Sure, that sounds reasonable.

If I understand what you mean then yes, it's a hit test in the sense that the structure item printed needs to have it, but nothing else inside of the structure item technically needs to have the cursor for it to hit.

Reply:
https://en.wikipedia.org/wiki/Hit-testing

Reply:
Ahhh yes. Now I understand. The answer is yes, it's a hit test, and everything containing the cursor has it, not just a single node.